components/org.wso2.micro.integrator.ndatasource.core/src/main/java/org/wso2/micro/integrator/ndatasource/core/utils/DataSourceUtils.java

```diff
@@ -244,6 +251,46 @@ public static void secureResolveDocument(Document doc, boolean checkSecureVault)
 		}
     }
 
+	public static void secureResolveOMElement(OMElement doc, boolean checkSecureVault)
+			throws DataSourceException {
+
+		if (doc != null) {
+			secretResolver = SecretResolverFactory.create(doc, true);
+			secureLoadOMElement(doc);
+		}
+	}
+
+	private static void secureLoadOMElement(OMElement element) {
+
+		String alias = MiscellaneousUtil.getProtectedToken(element.getText());
+		if (alias != null && !alias.isEmpty()) {
+			element.setText(loadFromSecureVault(alias));
+		} else {
+			OMAttribute secureAttr = element.getAttribute(new QName(DataSourceConstants.SECURE_VAULT_NS,
+					DataSourceConstants.SECRET_ALIAS_ATTR_NAME));
+			if (secureAttr != null) {
+				element.setText(loadFromSecureVault(secureAttr.getAttributeValue()));
+				element.removeAttribute(secureAttr);
+			}
+		}
+		Iterator<OMElement> childNodes = element.getChildElements();
+		while (childNodes.hasNext()) {
+			OMElement tmpNode = childNodes.next();
+			secureLoadOMElement(tmpNode);
+		}
+	}
+
+	public static OMElement convertToOMElement(File file) throws DataSourceException {
+
+		try {
+			OMElement documentElement = new StAXOMBuilder(new FileInputStream(file)).getDocumentElement();
+			return documentElement;
+		} catch (Exception e) {
+			throw new DataSourceException("Error in creating an XML document from file: " +
+					e.getMessage(), e);
```

Comment:
It has been added for the sake of finishing the sentence.

Reply:
Can't we just end the sentence with the file name (file.getAbsolutePath()) and let the detailed message e.message() be there in the exception itself. I think it is not necessary to log this separately whenever we are passing the exception along with the DataSourceException object because it will anyway gets printed with the exception. WDYT?